src/main/java/seedu/address/model/student/UniqueStudentList.java

```diff
@@ -128,14 +128,17 @@ public boolean hasName(Name name) {
      * student with the {@code name}
      * guarantees that student will not be null
      */
-    public boolean hasSession(Name name, Session session) {
-        requireAllNonNull(session, name);
-        Student student = getStudentWithName(name);
-        List<Session> sessionList = student.getListOfSessions();
-        SessionDate sessionDate = session.getSessionDate();
-        for (Session existingSession : sessionList) {
-            if (existingSession.getSessionDate().equals(sessionDate)) {
-                return true;
+    public boolean hasSession(Session target) {
+        requireNonNull(target);
+
+        for (Student student : internalList) {
+            List<Session> sessionList = student.getListOfSessions();
+            for (Session session : sessionList) {
+                SessionDate sessionDate = session.getSessionDate();
+                SessionDate targetSessionDate = target.getSessionDate();
+                if (sessionDate.equals(targetSessionDate)) {
+                    return true;
```

Comment:
Correct me if i'm wrong, is this a particular coding standard I do not know of?

Reply:
I guess i could improve it by bringing the targetSessioDate in the outer loop, since there isn't a need to declare the targetSessionDate in every iteration.